### PR TITLE
usb: remove Kconfig option USB_UART_CONSOLE

### PR DESCRIPTION
**Pull request by jfischer-no**
Adjust init level and priority of CDC ACM driver to regular serial drivers.
Limit scope of CONFIG_USB_UART_CONSOLE and finally remove this option.

Branched off from #40220

### CODE REVIEW

**Review by jhedberg — APPROVED** [review]

File: tests/bluetooth/shell/src/main.c [hunk 1]
```diff
@@ -122,10 +123,20 @@ static void hrs_notify(void)
 
 void main(void)
 {
-	if (IS_ENABLED(CONFIG_USB_UART_CONSOLE)) {
-		usb_enable(NULL);
-		k_sleep(K_SECONDS(2));
+#if DT_NODE_HAS_COMPAT(DT_CHOSEN(zephyr_shell_uart), zephyr_cdc_acm_uart)
+	const struct device *dev;
```
Comment (nordic-krch) [inline tests/bluetooth/shell/src/main.c:127]: what about moving this to shell backend. There is a `enable` function called from shell thread context. You can block there.
https://github.com/zephyrproject-rtos/zephyr/blob/525fa76f4dd13d42549261c9fc846a150ee291bc/subsys/shell/backends/shell_uart.c#L238

Comment (nordic-krch) [reply tests/bluetooth/shell/src/main.c:127]: However, that will handle shell only and not printk :/ maybe it must stay here or we redirect printk to shell like log is redirected.

Comment (jfischer-no) [reply tests/bluetooth/shell/src/main.c:127]: I would like to keep it that way for now. That is consistent with other examples. Also `CONFIG_SHELL_BACKEND_SERIAL_CHECK_DTR` is enabled, waiting here is general for this sample.

**Review by anangl — APPROVED** [review]
LGTM. Perhaps it would be good to correct the typo "sampes" in the a0c0996e0acf1bedfe297f9a6c8712ec3b66d39e commit title.

**Review by nordic-krch — APPROVED** [review]

**Review by carlescufi — APPROVED** [review]

**Pull request merged**